src/app/data-planes/data/index.ts

```diff
@@ -57,6 +66,50 @@ export type DataplaneOverview = PartialDataplaneOverview & {
   services: string[]
 }
 
+export type PolicyTypeEntryConnection = {
+  sourceTags: LabelValue[]
+  destinationTags: LabelValue[]
+  name: string | null
+  origins: Meta[]
+  config: string | undefined
+}
+
+export type PolicyTypeEntry = {
+  type: string
+  connections: PolicyTypeEntryConnection[]
+}
+
+export type SidecarDataplane = PartialSidecarDataplane
+
+export type RuleEntryRule = {
+  config: object | undefined
+  matchers?: InspectRuleMatcher[]
+  origins: Meta[]
+}
+
+export type RuleEntry = {
+  type: string
+  rules: RuleEntryRule[]
+}
+
+export type InspectRulesForDataplane = PartialInspectRulesForDataplane & {
+  proxyRule: RuleEntry | undefined
+  toRules: RuleEntry[]
+  fromRuleInbounds: Array<{ port: number, ruleEntries: RuleEntry[] }>
+}
+
+export type MeshGatewayListenerEntry = {
+  protocol: string
+  port: number
+  hostName: string
+  routeEntries: MeshGatewayRouteEntry[]
+}
+
+export type MeshGatewayDataplane = PartialMeshGatewayDataplane & {
+  listenerEntries: MeshGatewayListenerEntry[]
+  routePolicies: Meta[]
+}
+
```

Comment:
I wondered why these types are in `data` not `@/types`. Umm, question, if we had auto-generated types would these ones be auto generated?

Reply:
Good question. I changed this a bit so that the data file only defines/exports types it governs (i.e. the transformed types). In other words: the things that would be auto-generated (hopefully anyway) are back in the types file.

Reply:
Awesome, sounds great!